src/llvm/analysis/DefUse.cpp

```diff
@@ -331,7 +332,8 @@ void LLVMDefUseAnalysis::addDataDependence(LLVMNode *node, PSNode *pts,
         if (!defs.empty()) {
             for (RDNode *rd : defs) {
                 assert(!rd->isUnknown() && "Unknown memory defined at unknown location?");
-                addDataDependence(node, rd);
+                if (rd->getType() != RDNodeType::PHI)
```

Comment:
`getReachindDefinitions` returns also PHI nodes? Wouldn't it be better if it returned only the real definitions?

Reply:
I thought PHI nodes were returned too, but as I'm looking at [`SemisparseRda::merge_maps`](https://github.com/tomsik68/dg/blob/ef9122d4b67bca015b51b95058b045f746aa3cd4/src/analysis/ReachingDefinitions/SemisparseRda.h#L20), it's no longer the case. So this check could be removed. 

However, it could be useful to have an assertion that `rd` is associated with an `llvm::Value`, as I remember seeing nasty errors when I used artificial PHI nodes without `llvm::Value`.